base/sort.jl

```diff
@@ -1739,12 +1739,16 @@ function sort!(v::AbstractVector{T};
 end
 
 """
-    sort(v::Union{AbstractVector, NTuple}; alg::Base.Sort.Algorithm=Base.Sort.defalg(v), lt=isless, by=identity, rev::Bool=false, order::Base.Order.Ordering=Base.Order.Forward)
+    sort(v; alg::Base.Sort.Algorithm=Base.Sort.defalg(v), lt=isless, by=identity, rev::Bool=false, order::Base.Order.Ordering=Base.Order.Forward)
+    sort(v::NTuple; kws...) -> NTuple
```

Comment:
```suggestion
    sort(v::NTuple; kws...)::NTuple
```

I prefer `::` for declaring the return type, in fact I think the `->` may be very confusing when used for this purpose, because:
* `->` is actual Julia syntax for defining functions, with conflicting semantics (defines a function with args on the LHS and return value on RHS)
* `::` is actual Julia syntax, too, but with semantics as expected
* not sure, but I think the `->` is sometimes used in doc strings to express the returned value, as opposed to the type

Reply:
Agreed!

Reply:
Taking this to the next level: https://github.com/JuliaLang/julia/pull/57012